[charts] Add sparkline (#9662)
mui/mui-x: docs/data/charts/sparkline/sparkline.md

---
title: Charts - Sparkline
---

# Charts - Sparkline

<p class="description">Sparkline charts can provide an overview of data trends.</p>

## Basics

> 

Reviewer: I feel we could add a brief explanation for it.
```suggestion

A sparkline is a small chart drawn without axes or coordinates to present the general shape of a variation in a simplified way.
```

The `<SparklineCharts />` requires only the `data` props which is an array of numbers.

> You can also choose between spark ann line plot with `plotType="bar"`.

Reviewer: ```suggestion
You can also switch from line to a bar plot with `plotType="bar"`.
```

{{"demo": "BasicSparkLine.js"}}

## Line customization

You can fill the area below the sparkline curve with `area` props.

> You can also modify the interpolation with `curve` props, as explained in the [line charts docs](/x/react-charts/lines/#interpolation).

Reviewer: ```suggestion
To modify the interpolation with `curve` props, follow the instructions in the [line charts docs](/x/react-charts/lines/#interpolation).
```

{{"demo": "AreaSparkLine.js"}}

## Interaction

> Compared to line and bar charts, the spark-line chart has some additional props to simplify interaction configuration.

Reviewer: ```suggestion
Compared to line and bar charts, the sparkline chart has some additional props to simplify interaction configuration.
```

> You can use `showTooltip` and `showHighlight` to add the default tooltip and highlight to your spark-line component.

Reviewer: ```suggestion
You can use `showTooltip` and `showHighlight` to display the default tooltip and highlight in your sparkline.
```

Those are helpers.

> If you need more advanced customization, you can still provide the tooltip and highlight props described on the [dedicated docs page](/x/react-charts/tooltip/).

Reviewer: ```suggestion
If you need more advanced customization, you can provide custom props for `tooltip` and `highlight` as described in the [Tooltip page](/x/react-charts/tooltip/).
```

{{"demo": "BasicSparkLineCustomization.js"}}

## Axis management

> By default, the sparkline will set x values to 0, 1, 2, ... and hide it in the tooltip.

Reviewer: ```suggestion
By default, the sparkline assigns `xAxis` values as an ascending integer sequence starting from 0 (0, 1, 2,...).  These values are, in this case, hidden in the tooltip.
```

> You can override this behavior if your data are not evenly distributed, or if their value matter.

Reviewer: ```suggestion
You can override this behavior if your data are not evenly distributed, or if you need to label them.
```

Should we also push a bit to showcase an evenly distributed data?

<img width="412" alt="image" src="https://github.com/mui/mui-x/assets/550141/9a8142c6-0753-4fad-90d7-1fa0ee4577b9">
